tests/test_exp_typed_attn_res.py

```
@@ -0,0 +1,667 @@
+    expected = dict(reference.named_parameters())
+    actual = dict(checkpointed.named_parameters())
+    assert expected.keys() == actual.keys()
+    for name in expected:
+        if expected[name].grad is not None:
+            assert actual[name].grad is not None
+            torch.testing.assert_close(
+                expected[name].grad,
+                actual[name].grad,
+                rtol=0,
+                atol=0,
+                msg=lambda message: f"{name}: {message}",
```

Ruff (0.16.0)

[warning] Function definition does not bind loop variable `name`

(B023)